### ISSUE TITLE
VCell 7.3.0.13 appears to be broken

### ISSUE THREAD

**Issue by jonrkarr** [biosimulators/Biosimulators_VCell#60]
Below is the output from executing the VCell 7.3.0.13 Docker image with the Vilar example. Other archives generate similar issues

- No permission to write to the log `jsbml.log`
- Unable to find `.ida` files
- HDF5 conversion appears to fail because python3 isn't installed
- Status appears not to be logged because python3 isn't installed

The HDF5 and status issues appear to be new since 7.3.0.13. The `jsbml.log` and `.ida` issues have been present at least since 7.3.0.07.

```
VCell CLI input archive /tmp/in/Vilar-PNAS-2002-minimal-circardian-clock.omex
2021-02-27 19:10:14,503 main ERROR FileManager (jsbml.log) java.io.FileNotFoundException: jsbml.log (Permission denied) java.io.FileNotFoundException: jsbml.log (Permission denied)
	at java.io.FileOutputStream.open0(Native Method)
	at java.io.FileOutputStream.open(FileOutputStream.java:270)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
	at org.apache.logging.log4j.core.appender.FileManager$FileManagerFactory.createManager(FileManager.java:290)
	at org.apache.logging.log4j.core.appender.FileManager$FileManagerFactory.createManager(FileManager.java:274)
	at org.apache.logging.log4j.core.appender.AbstractManager.getManager(AbstractManager.java:112)
	at org.apache.logging.log4j.core.appender.OutputStreamManager.getManager(OutputStreamManager.java:114)
	at org.apache.logging.log4j.core.appender.FileManager.getFileManager(FileManager.java:112)
	at org.apache.logging.log4j.core.appender.FileAppender$Builder.build(FileAppender.java:86)
	at org.apache.logging.log4j.core.appender.FileAppender$Builder.build(FileAppender.java:51)
	at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.build(PluginBuilder.java:122)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createPluginObject(AbstractConfiguration.java:952)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:892)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:884)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.doConfigure(AbstractConfiguration.java:508)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.initialize(AbstractConfiguration.java:232)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.start(AbstractConfiguration.java:244)
	at org.apache.logging.log4j.core.LoggerContext.setConfiguration(LoggerContext.java:545)
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:617)
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:634)
	at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:229)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:152)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:45)
	at org.apache.logging.log4j.LogManager.getContext(LogManager.java:194)
	at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:551)
	at cbit.vcell.util.NativeLoader.<clinit>(NativeLoader.java:65)
	at cbit.vcell.resource.OperatingSystemInfo.<init>(OperatingSystemInfo.java:182)
	at cbit.vcell.resource.OperatingSystemInfo.<init>(OperatingSystemInfo.java:22)
	at cbit.vcell.resource.OperatingSystemInfo$OSIHolder.<clinit>(OperatingSystemInfo.java:204)
	at cbit.vcell.resource.OperatingSystemInfo.getInstance(OperatingSystemInfo.java:73)
	at org.vcell.cli.CLIUtils.<clinit>(CLIUtils.java:35)
	at org.vcell.cli.CLIStandalone.singleExec(CLIStandalone.java:62)
	at org.vcell.cli.CLIStandalone.main(CLIStandalone.java:42)

2021-02-27 19:10:14,510 main ERROR Unable to inject fields into builder class for plugin type class org.apache.logging.log4j.core.appender.FileAppender, element File. java.lang.IllegalStateException: ManagerFactory [org.apache.logging.log4j.core.appender.FileManager$FileManagerFactory@15b204a1] unable to create manager for [jsbml.log] with data [org.apache.logging.log4j.core.appender.FileManager$FactoryData@77167fb7]
	at org.apache.logging.log4j.core.appender.AbstractManager.getManager(AbstractManager.java:114)
	at org.apache.logging.log4j.core.appender.OutputStreamManager.getManager(OutputStreamManager.java:114)
	at org.apache.logging.log4j.core.appender.FileManager.getFileManager(FileManager.java:112)
	at org.apache.logging.log4j.core.appender.FileAppender$Builder.build(FileAppender.java:86)
	at org.apache.logging.log4j.core.appender.FileAppender$Builder.build(FileAppender.java:51)
	at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.build(PluginBuilder.java:122)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createPluginObject(AbstractConfiguration.java:952)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:892)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:884)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.doConfigure(AbstractConfiguration.java:508)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.initialize(AbstractConfiguration.java:232)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.start(AbstractConfiguration.java:244)
	at org.apache.logging.log4j.core.LoggerContext.setConfiguration(LoggerContext.java:545)
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:617)
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:634)
	at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:229)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:152)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:45)
	at org.apache.logging.log4j.LogManager.getContext(LogManager.java:194)
	at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:551)
	at cbit.vcell.util.NativeLoader.<clinit>(NativeLoader.java:65)
	at cbit.vcell.resource.OperatingSystemInfo.<init>(OperatingSystemInfo.java:182)
	at cbit.vcell.resource.OperatingSystemInfo.<init>(OperatingSystemInfo.java:22)
	at cbit.vcell.resource.OperatingSystemInfo$OSIHolder.<clinit>(OperatingSystemInfo.java:204)
	at cbit.vcell.resource.OperatingSystemInfo.getInstance(OperatingSystemInfo.java:73)
	at org.vcell.cli.CLIUtils.<clinit>(CLIUtils.java:35)
	at org.vcell.cli.CLIStandalone.singleExec(CLIStandalone.java:62)
	at org.vcell.cli.CLIStandalone.main(CLIStandalone.java:42)

2021-02-27 19:10:14,514 main ERROR Unable to invoke factory method in class class org.apache.logging.log4j.core.appender.FileAppender for element File. java.lang.IllegalStateException: No factory method found for class org.apache.logging.log4j.core.appender.FileAppender
	at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.findFactoryMethod(PluginBuilder.java:224)
	at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.build(PluginBuilder.java:130)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createPluginObject(AbstractConfiguration.java:952)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:892)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:884)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.doConfigure(AbstractConfiguration.java:508)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.initialize(AbstractConfiguration.java:232)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.start(AbstractConfiguration.java:244)
	at org.apache.logging.log4j.core.LoggerContext.setConfiguration(LoggerContext.java:545)
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:617)
	at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:634)
	at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:229)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:152)
	at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:45)
	at org.apache.logging.log4j.LogManager.getContext(LogManager.java:194)
	at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:551)
	at cbit.vcell.util.NativeLoader.<clinit>(NativeLoader.java:65)
	at cbit.vcell.resource.OperatingSystemInfo.<init>(OperatingSystemInfo.java:182)
	at cbit.vcell.resource.OperatingSystemInfo.<init>(OperatingSystemInfo.java:22)
	at cbit.vcell.resource.OperatingSystemInfo$OSIHolder.<clinit>(OperatingSystemInfo.java:204)
	at cbit.vcell.resource.OperatingSystemInfo.getInstance(OperatingSystemInfo.java:73)
	at org.vcell.cli.CLIUtils.<clinit>(CLIUtils.java:35)
	at org.vcell.cli.CLIStandalone.singleExec(CLIStandalone.java:62)
	at org.vcell.cli.CLIStandalone.main(CLIStandalone.java:42)

2021-02-27 19:10:14,515 main ERROR Null object returned for File in Appenders.
2021-02-27 19:10:14,529 main ERROR Unable to locate appender "file" for logger config "root"
2021-02-27 19:10:14,529 main ERROR Unable to locate appender "file" for logger config "org.sbml"
Feb 27, 2021 7:10:14 PM java.util.prefs.FileSystemPreferences$1 run
WARNING: Couldn't create user preferences directory. User preferences are unusable.
Feb 27, 2021 7:10:14 PM java.util.prefs.FileSystemPreferences$1 run
WARNING: java.io.IOException: No such file or directory
-----------------------------------------------------------------------------------------------------
Successful translation: SED-ML file simulation.sedml
-----------------------------------------------------------------------------------------------------
Task 'tsk_0_0' is compatible, solver match found in ontology: 'KISAO:0000019' matched to CVODE
Succesful model import: SBML file /tmp/temp2421917748719536134.xml
Task 'tsk_0_1' is compatible, solver match found in ontology: 'KISAO:0000019' matched to CVODE
Succesful model import: SBML file /tmp/temp8255427163444252271.xml
Task 'tsk_0_2' is compatible, solver match found in ontology: 'KISAO:0000030' matched to ForwardEuler
Task 'tsk_0_3' is compatible, solver match found in ontology: 'KISAO:0000030' matched to ForwardEuler
Task 'tsk_1_0' is compatible, solver match found in ontology: 'KISAO:0000027' matched to StochGibson
Succesful model import: SBML file /tmp/temp7204919949622790549.xml
Task 'tsk_1_1' is compatible, solver match found in ontology: 'KISAO:0000027' matched to StochGibson
Succesful model import: SBML file /tmp/temp6078124204231790672.xml
java.io.FileNotFoundException: /tmp/out/simulation.sedml/SimID_686827113_0_.ida (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at cbit.vcell.solver.ode.SundialsSolver.getStateVariableResultSet(SundialsSolver.java:299)
	at cbit.vcell.solver.ode.SundialsSolver.getODESolverResultSet(SundialsSolver.java:187)
	at org.vcell.cli.SolverHandler.simulateAllTasks(SolverHandler.java:90)
	at org.vcell.cli.CLIStandalone.singleExec(CLIStandalone.java:104)
	at org.vcell.cli.CLIStandalone.main(CLIStandalone.java:42)
Solver status: 3
Solver message: ./.vcell/simdata/temp/686827113_link/SundialsSolverStandalone_x64
Failed execution: Model 'Vilar-PNAS-2002-minimal-circardian-clock_ODE_deterministic' Task 'tsk_0_0'.
-----------------------------------------------------------------------------------------------------
Solver status: 3
Solver message: simulation failed at time=1.3: consider using smaller default or max time step.
values are:
A = NaN
C = Infinity
PrmA = Infinity
PrmA_bound = -Infinity
PrmR = Infinity
PrmR_bound = -Infinity
R = -Infinity

Failed execution: Model 'Vilar-PNAS-2002-minimal-circardian-clock_ODE_deterministic' Task 'tsk_0_2'.
-----------------------------------------------------------------------------------------------------
final tolerance=0.1 final threshold=0.09, 885 remaining (keepAtMost=1000)
Succesful execution: Model 'Vilar-PNAS-2002-minimal-circardian-clock_ODE_deterministic' Task 'tsk_0_3'.
-----------------------------------------------------------------------------------------------------
java.io.FileNotFoundException: /tmp/out/simulation.sedml/SimID_35662342_0_.ida (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at cbit.vcell.solver.ode.SundialsSolver.getStateVariableResultSet(SundialsSolver.java:299)
	at cbit.vcell.solver.ode.SundialsSolver.getODESolverResultSet(SundialsSolver.java:187)
	at org.vcell.cli.SolverHandler.simulateAllTasks(SolverHandler.java:90)
	at org.vcell.cli.CLIStandalone.singleExec(CLIStandalone.java:104)
	at org.vcell.cli.CLIStandalone.main(CLIStandalone.java:42)
Solver status: 3
Solver message: ./.vcell/simdata/temp/35662342_link/SundialsSolverStandalone_x64
Failed execution: Model 'Vilar-PNAS-2002-minimal-circardian-clock_ODE_deterministic_0' Task 'tsk_0_1'.
-----------------------------------------------------------------------------------------------------
java.io.FileNotFoundException: /tmp/out/simulation.sedml/SimID_1708654737_0_.ida (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at cbit.vcell.solver.stoch.GibsonSolver.getStochSolverResultSet(GibsonSolver.java:121)
	at cbit.vcell.solver.stoch.GibsonSolver.printStochFile(GibsonSolver.java:273)
	at cbit.vcell.solver.stoch.GibsonSolver.cleanup(GibsonSolver.java:64)
	at cbit.vcell.solvers.AbstractCompiledSolver.runSolver(AbstractCompiledSolver.java:183)
	at org.vcell.cli.SolverHandler.simulateAllTasks(SolverHandler.java:88)
	at org.vcell.cli.CLIStandalone.singleExec(CLIStandalone.java:104)
	at org.vcell.cli.CLIStandalone.main(CLIStandalone.java:42)
java.io.FileNotFoundException: /tmp/out/simulation.sedml/SimID_1708654737_0_.ida (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at cbit.vcell.solver.stoch.GibsonSolver.getStochSolverResultSet(GibsonSolver.java:121)
	at org.vcell.cli.SolverHandler.simulateAllTasks(SolverHandler.java:92)
	at org.vcell.cli.CLIStandalone.singleExec(CLIStandalone.java:104)
	at org.vcell.cli.CLIStandalone.main(CLIStandalone.java:42)
Solver status: 3
Solver message: ./.vcell/simdata/temp/1708654737_link/VCellStoch_x64
Failed execution: Model 'Vilar-PNAS-2002-minimal-circardian-clock_ODE_stochastic' Task 'tsk_1_0'.
-----------------------------------------------------------------------------------------------------
java.io.FileNotFoundException: /tmp/out/simulation.sedml/SimID_1891293665_0_.ida (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at cbit.vcell.solver.stoch.GibsonSolver.getStochSolverResultSet(GibsonSolver.java:121)
	at cbit.vcell.solver.stoch.GibsonSolver.printStochFile(GibsonSolver.java:273)
	at cbit.vcell.solver.stoch.GibsonSolver.cleanup(GibsonSolver.java:64)
	at cbit.vcell.solvers.AbstractCompiledSolver.runSolver(AbstractCompiledSolver.java:183)
	at org.vcell.cli.SolverHandler.simulateAllTasks(SolverHandler.java:88)
	at org.vcell.cli.CLIStandalone.singleExec(CLIStandalone.java:104)
	at org.vcell.cli.CLIStandalone.main(CLIStandalone.java:42)
java.io.FileNotFoundException: /tmp/out/simulation.sedml/SimID_1891293665_0_.ida (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at cbit.vcell.solver.stoch.GibsonSolver.getStochSolverResultSet(GibsonSolver.java:121)
	at org.vcell.cli.SolverHandler.simulateAllTasks(SolverHandler.java:92)
	at org.vcell.cli.CLIStandalone.singleExec(CLIStandalone.java:104)
	at org.vcell.cli.CLIStandalone.main(CLIStandalone.java:42)
Solver status: 3
Solver message: ./.vcell/simdata/temp/1891293665_link/VCellStoch_x64
Failed execution: Model 'Vilar-PNAS-2002-minimal-circardian-clock_ODE_stochastic_0' Task 'tsk_1_1'.
-----------------------------------------------------------------------------------------------------
Failed executing the command: `python3 --version`

Check Python installation...
Failed executing the command: `python3 --version`

Check Python installation...
Failed installing PIP packages.
Failed executing the command: `python3 --version`

Check Python installation...
Ignoring unsupported output plot2d_Fig_1a
Ignoring unsupported output plot2d_low_delta_R_det
Ignoring unsupported output plot2d_Euler_default
Ignoring unsupported output plot2d_Euler_small_step_size
Ignoring unsupported output plot2d_Fig_1_c
Ignoring unsupported output plot2d_low_delta_R_stoch
Failed executing the command: `python3 --version`

Check Python installation...
HDF5 conversion failed...
Failed executing the command: `python3 --version`

Check Python installation...
Failed generating status YAML...
Failed executing the command: `python3 --version`

Check Python installation...
Failed generating status YAML...
```

**Comment by gmarupilla** [comment]
This commit https://github.com/virtualcell/vcell/commit/be75020293d31562c1550d30e6b1e92985ae5486 fixes the `jsbml.log` and `.ida` file issues. I'll look into the problem of HDF5 conversion and the status YAML.

**Comment by jonrkarr** [comment]
How are you generating the HDF and log? Did you write your own code, or are you using BioSimulators-utils. Either way, once we get VCell passing one test, VCell should pass most of the tests.

I'll update the entry for VCell in the BioSimulators database with the current specifications in this repo.

**Comment by jonrkarr** [comment]
I see you're installing biosimulators-utils in the Docker image. I assume you're using this for the HDF5 and logs.

Regarding the python3 issue, I can see that python3 is installed into the Docker image. This makes me think that python3 isn't in the system path that the Java cli program is using. Have you looked at the system path that Java is using? Maybe python3 (`/usr/local/bin/python3`) needs to be included in the system path.

**Comment by moraru** [comment]
I believe this to be true.  But @gmarupilla should confirm, he coded that part.

**Comment by gmarupilla** [comment]
> I see you're installing biosimulators-utils in the Docker image. I assume you're using this for the HDF5 and logs.
> 
> Regarding the python3 issue, I can see that python3 is installed into the Docker image. This makes me think that python3 isn't in the system path that the Java cli program is using. Have you looked at the system path that Java is using? Maybe python3 (`/usr/local/bin/python3`) needs to be included in the system path.

Yes, I wrote a python command-line utility package that has the dependency of biosimulators_utils. This script runs in JAVA run time environment for HDF5 conversion from CSV and status YAML generation. It appears that I need to add a PYTHONPATH to the docker image.

**Comment by gmarupilla** [comment]
@jonrkarr I wasn't able to reproduce the same issue from my Linux or Mac machine. However, I'm getting the same error(s) while executing the simulation in singularity image #57. I tried executing the shell in the singularity image as well and then tried to access python from the singularity shell, I was able to execute python commands from the singularity shell, but it wasn't showing up in Java run time (only in singularity, but works fine in docker). Do the above logs from the docker run from GHCR?
I tried in my local pulling the image from GHCR, below is the screenshot.
![image](https://user-images.githubusercontent.com/53095348/109544440-93018c00-7a95-11eb-85e5-50208084ceeb.png)

**Comment by gmarupilla** [comment]
I can reproduce the issue and found the root cause(after working on a physical Linux machine). There are few process builders in Java code merely for the Linux system, which some needs to be included in the System properties. I'm working on it. I'll push the fix.

**Comment by jonrkarr** [comment]
Great!

**Comment by gmarupilla** [comment]
Fixed in #61
